java/src/main/java/com/powsybl/dataframe/network/extensions/SlackTerminalDataframeProvider.java

```diff
@@ -0,0 +1,48 @@
+package com.powsybl.dataframe.network.extensions;
+
+import com.google.auto.service.AutoService;
+import com.powsybl.dataframe.network.NetworkDataframeMapper;
+import com.powsybl.dataframe.network.NetworkDataframeMapperBuilder;
+import com.powsybl.dataframe.network.adders.NetworkElementAdder;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.SlackTerminal;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * @author Etienne Lesot <etienne.lesot@rte-france.com>
+ */
+@AutoService(NetworkExtensionDataframeProvider.class)
+public class SlackTerminalDataframeProvider implements NetworkExtensionDataframeProvider {
+    @Override
+    public String getExtensionName() {
+        return "slackTerminal";
+    }
+
+    private Stream<SlackTerminal> itemsStream(Network network) {
+        return network.getVoltageLevelStream()
+                .map(vl -> (SlackTerminal) vl.getExtension(SlackTerminal.class))
+                .filter(Objects::nonNull);
+    }
+
+    @Override
+    public NetworkDataframeMapper createMapper() {
+        return NetworkDataframeMapperBuilder.ofStream(this::itemsStream)
+                .stringsIndex("id", st -> st.getTerminal().getConnectable().getId())
+                .build();
+    }
+
+    @Override
+    public void removeExtensions(Network network, List<String> ids) {
```

Comment:
Maybe it's weird to expect voltage level IDs here, since in the returned dataframe we don't expose it.
So I think we should either:
- add the voltage_level_id as a new column
- accept here "element" ids and/or bus ids 

But in the second case it's not really clear what is expected, so I propose to keep the voltage_level_id solution ...